lib/tasks/data/remove_transient_data.rb

```diff
@@ -12,8 +13,34 @@ class Session < ActiveRecord::Base
   def call
     Rails.logger.info("#{self.class.name}: processing")
 
-    Spree::StateChange.where("created_at < ?", RETENTION_PERIOD).delete_all
-    Spree::LogEntry.where("created_at < ?", RETENTION_PERIOD).delete_all
-    Session.where("updated_at < ?", RETENTION_PERIOD).delete_all
+    Spree::StateChange.where("created_at < ?", MEDIUM_RETENTION).delete_all
+    Spree::LogEntry.where("created_at < ?", MEDIUM_RETENTION).delete_all
+    Session.where("updated_at < ?", SHORT_RETENTION).delete_all
+
+    clear_old_cart_data!
+  end
+
+  private
+
+  def clear_old_cart_data!
+    old_carts = Spree::Order.
+      where("spree_orders.state = 'cart' AND spree_orders.updated_at < ?", SHORT_RETENTION).
+      merge(orders_without_payments)
+
+    old_cart_line_items = Spree::LineItem.where(order_id: old_carts)
```

Comment:
have you checked the resulting SQL statement? I wonder if we don't hit any query char limit or similar. You said `old_carts` can be by the order of a million in some cases, right?